rkt/image/httpfetcher.go

```diff
@@ -120,28 +124,37 @@ func (f *httpFetcher) fetchVerifiedURL(u *url.URL, a *asc, etag string) (readSee
 	f.maybeOverrideAscFetcherWithRemote(o, u, a)
 	ascFile, retry, err := o.DownloadSignature(a)
 	if err != nil {
-		return nil, nil, err
+		return NopReadSeekCloser(nil), nil, err
 	}
-	defer func() { maybeClose(ascFile) }()
+	defer func() {
+		if ascFile != nil {
+			ascFile.Close()
+		}
+	}()
```

Comment:
Since `DownloadSignature` now returns in case of non-err a closeable `readSeekCloser` via [`*asc.Get`](https://github.com/coreos/rkt/pull/3120/files#diff-bbca85443e1d6dedcff42f176ad118a9R84) we can just `defer ascFile.Close()`